File: web/src/main/java/no.nav.sbl/rest/domain/RSEvent.java
package no.nav.sbl.rest.domain;

public class RSEvent {

    public long id;
    public String veilederIdent;
    public String eventType;

    public RSEvent withId(long id) {
        this.id = id;
        return this;
    }

    public RSEvent withVeilederIdent(String veilederIdent) {
        this.veilederIdent = veilederIdent;
        return this;
    }

    public RSEvent withEventType(String eventType) {
        this.eventType = eventType;
        return this;
    }
}

File: src/main/java/no/nav/sbl/rest/model/RSEvent.kt
package no.nav.sbl.rest.model

import no.nav.sbl.domain.VeilederContext

data class RSEvent(
    val veilederIdent: String,
    val eventType: String,
) {
    companion object {
        fun from(veilederContext: VeilederContext) =
            RSEvent(
                veilederIdent = veilederContext.veilederIdent,
                eventType = veilederContext.contextType.name,
            )
    }
}
